### PR TITLE
manage sources

### PR DESCRIPTION
**Pull request by BertrandRenault**
## Issue

Sources were not managed at all in front

Fixes https://github.com/pixano/pixano-project-manager/issues/76

## Description

Correctly set sources, dispatch entities in inspector according to sources
Annotations created within pixano app are sourced with a common Pixano source (created and saved if not present)
{id: "pixano_source", kind: "other", name: "Pixano"}

### CODE REVIEW

File: ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte [hunk 1]
```diff
@@ -266,7 +273,7 @@ License: CECILL-C
     } else {
       //updated an interpolated annotation: create it
       //use start ann of interpolated as base for new ann
-      let new_ann: Annotation;
+      let new_ann: Annotation | undefined = undefined;
```
Comment (juliendenize) [inline ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte:276]: Refactor to CamelCase

File: ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte [hunk 2]
```diff
@@ -257,6 +262,8 @@ License: CECILL-C
           `ERROR: mismatching types ${shape.type} & ${update_ann.table_info.base_schema}`,
         );
       }
+      const pix_source = getPixanoSource(sourcesStore);
+      update_ann.data.source_ref = { id: pix_source.id, name: pix_source.table_info.name };
```
Comment (juliendenize) [inline ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte:266]: Refactor to CamelCase

File: ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte [hunk 3]
```diff
@@ -306,8 +313,13 @@ License: CECILL-C
...
         //mask not implemented yet in video
       }
+      if (!new_ann) {
+        //TODO - remove this when mask managed (used mainly to avoid lint warnings)
+        throw new Error("Masks are not managed yet in video!");
+      }
       //update
-      //TODO note: lint warnings because KPT and mask not covered yet. (new_ann not set in these cases))
+      const pix_source = getPixanoSource(sourcesStore);
+      new_ann.data.source_ref = { id: pix_source.id, name: pix_source.table_info.name };
```
Comment (juliendenize) [inline ui/components/datasetItemWorkspace/src/components/DatasetItemViewer/VideoViewer.svelte:322]: Refactor to CamelCase

File: ui/components/datasetItemWorkspace/src/components/Inspector/ObjectCard.svelte [hunk 4]
```diff
@@ -50,10 +50,18 @@ License: CECILL-C
 
   let open: boolean = false;
   let showIcons: boolean = false;
+  let isHighlighted: boolean = false;
+  let isEditing: boolean = false;
+  let isVisible: boolean = true;
+  let boxIsVisible: boolean = true;
+  let maskIsVisible: boolean = true;
+  let keypointsIsVisible: boolean = true;
+  let display_name: string = `${entity.data.name ? (entity.data.name as string) + " - " : ""}${entity.id}`;
```
Comment (juliendenize) [inline ui/components/datasetItemWorkspace/src/components/Inspector/ObjectCard.svelte:59]: Refactor to CamelCase

**Review by juliendenize — CHANGES_REQUESTED** [review]
I put comments to refactor to CamelCase. I know that a lot of places are not respecting this but we should not add more and perform a pass to normalize everything at some point.

If I understood correctly, everytime you change a bbox it will be a pixano source ? Indeed for now we don't manage users (maybe never ?) so I guess it is a viable solution until we decide to move forward. I just didn't get one thing, is it everytime or is it only if the source is a model / groud_truth / other and/or human ?

**Comment by BertrandRenault** [comment]
> If I understood correctly, everytime you change a bbox it will be a pixano source ? Indeed for now we don't manage users (maybe never ?) so I guess it is a viable solution until we decide to move forward. I just didn't get one thing, is it everytime or is it only if the source is a model / groud_truth / other and/or human ?

It's every time. Whenever there is a change (or creation), "Pixano source" is applied (replace previous source when editing, set when creating)

**Comment by juliendenize** [comment]
> > If I understood correctly, everytime you change a bbox it will be a pixano source ? Indeed for now we don't manage users (maybe never ?) so I guess it is a viable solution until we decide to move forward. I just didn't get one thing, is it everytime or is it only if the source is a model / groud_truth / other and/or human ?
> 
> It's every time. Whenever there is a change (or creation), "Pixano source" is applied (replace previous source when editing, set when creating)

Looks good to me for now then. Maybe at some point we will allow the user to give themself a name so that they can mimic a multi-user approach but it might be out-of-scope for now.

**Pull request merged**